src/oemof_tabular_plugins/general/post_processing/post_processing.py

```diff
@@ -650,18 +656,29 @@ def create_costs_table(all_scalars, results, capacities_df, storage_capacities_d
     return costs_df
 
 
-def post_processing(params, results, results_path):
+class OTPCalculator(Calculator):
+    def __init__(self, input_parameters, energy_system, dp_path):
+        self.df_results = construct_dataframe_from_results(energy_system)
+        process_raw_results(self.df_results)
+        self.df_results = process_raw_inputs(self.df_results, dp_path)
+        apply_calculations(self.df_results)
+        super().__init__(input_parameters, energy_system.results)
+
+
+def post_processing(params, es, results_path, dp_path):
     # ToDo: adapt this function after multi-index dataframe is implemented to make it more concise / cleaner
     # ToDo: params can be accessed in results so will not need to be a separate argument
     """
     The main post-processing function extracts various scalar and timeseries data and stores it in CSV files.
     :param params: energy system parameters
-    :param results: oemof model results
+    :param es: oemof energy_system with results in it, ie es.results = processing.results(m) has been performed
     :param results_path: results directory path
+    :param dp_path: path to the datapackage.json file
     """
     # initiate calculator for post-processing
-    calculator = Calculator(params, results)
-
+    calculator = OTPCalculator(params, es, dp_path)
+    print(calculator.df_results)
```

Comment:
@ciaradunks  this is just for testing purposes, you can set a breakpoint here to have a look at the multi-index dataframe